### PR TITLE
Add framework attributes to DLFW iterator tests.

### PR DESCRIPTION
**Pull request by mzient**
## Category:
**New feature** (*non-breaking change which adds functionality*)
**Refactoring** (*Redesign of existing code that doesn't affect functionality*)

## Description:
Prior to this change it was impossible to run the DLFW iterator tests only for selected frameworks. This PR makes it possible. There is also some refactoring for nose2, which doesn't requires attributes on test cases produced by test generators (`yield func, *params`).

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [X] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [X] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/test/python/test_fw_iterators.py [hunk 1]
```diff
@@ -441,6 +448,7 @@ def check_iterator_results(
     return (ids, sample_counter, per_gpu_counter, epoch_counter, rounded_shard_size)
 
 
+@attr("mxnet")
 def check_mxnet_iterator_pass_reader_name(
```
Comment (mzient) [inline dali/test/python/test_fw_iterators.py:452]: `yield check_mxnet_iterator_pass_reader_name, ...` caused nose2 to lose the attribute - it has to be added to this function, too - unlike for nosetests, which require it on the generator.

**Review by jantonguirao — APPROVED** [review]

File: qa/TL0_FW_iterators/test_mxnet.sh [hunk 2]
```diff
@@ -21,8 +21,7 @@ test_body() {
         done
     fi
     ${python_invoke_test} -m '(?:^|[\b_\./-])[Tt]est.*mxnet*' test_fw_iterators_detection.py
-    ${python_invoke_test} -m '(?:^|[\b_\./-])[Tt]est.*mxnet*' test_fw_iterators.py
-    ${python_invoke_test} -m '(?:^|[\b_\./-])[Tt]est.*gluon*' test_fw_iterators.py
+    ${python_new_invoke_test} -A 'mxnet,gluon' test_fw_iterators
```
Comment (stiepan) [inline qa/TL0_FW_iterators/test_mxnet.sh:24]: I looked into the L0_TL0_FW_iterators--fw-mxnet job and it seems no tests were run for mxnet. Can you double check if it picks them correctly?

Comment (JanuszL) [reply qa/TL0_FW_iterators/test_mxnet.sh:24]: There should be a separate job for that L0_TL0_FW_iterators--fw-mxnet

Comment (mzient) [reply qa/TL0_FW_iterators/test_mxnet.sh:24]: Nice catch! It seems that comma-separated attributes don't work in nose2 (again, nose2 sucks big time).

Comment (JanuszL) [reply qa/TL0_FW_iterators/test_mxnet.sh]: How about using expressions - https://docs.nose2.io/en/latest/plugins/attrib.html#select-tests-using-python-expressions ?

Comment (JanuszL) [reply qa/TL0_FW_iterators/test_mxnet.sh]: According to https://github.com/nose-devs/nose2/blob/main/nose2/plugins/attrib.py if you pass a list of tuple of attributes it should work as well.

Comment (mzient) [reply qa/TL0_FW_iterators/test_mxnet.sh]: I could, but is it really worth the complication? Using attributes (like in the updated code) works quite fine. My goal here was to simplify manual launches and especially filtering out unwanted frameworks. As I understand, it works like this:
```
-A 'a1,!a2' -A 'a3'
```
translates to `(a1 and not a2 ) or a3` - it makes perfect sense.

**Pull request merged**